docs/docs/0.4.0/documentation/secret-management.md

```diff
@@ -0,0 +1,134 @@
+# Secret Management
+
+Kuma lets you store sensitive information such as TLS keys, tokens etc. and reference them in Kuma policies.
+Every Secret stored in Kuma is in a scope of the Mesh.
+
+## Management
+
+### Universal
+
+The following snippet presents the format of Secret
+
+```yaml
+type: Secret
+name: sample-secret
+mesh: default
+data: dGVzdAo= # bytes encoded in Base64
+```
+
+Use `kumactl` to manage secret like any other Kuma resource on Universal.
+
+```sh
+$ cat secret.yaml
+type: Secret
+mesh: default
+name: sample-secret
+data: dGVzdAo=
+
+$ kumactl apply -f secret.yaml
+
+$ kumactl get secrets
+MESH      NAME
+default   sample-secret 
+```
+
+#### Access to the Secret API
+
+This API is exposed on Admin Server which means that by default it is only available on the same machine as the Control Plane.
+Consult [Accessing Admin Server from a different machine](security/#accessing-admin-server-from-a-different-machine) how to configure remote access.
+
+### Kubernetes
+
+On Kubernetes, Kuma leverages native [Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/) mechanism.
```

Comment:
"the native [Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/) resource"